specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-04-01-preview/workspaceRP.json

```diff
@@ -0,0 +1,6198 @@
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Sku"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "PrivateLinkResourceListResult": {
+      "description": "A list of private link resources",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          }
+        }
+      }
+    },
+    "PrivateLinkResourceProperties": {
+      "description": "Properties of a private link resource.",
+      "type": "object",
+      "properties": {
+        "groupId": {
+          "description": "The private link resource group id.",
+          "type": "string"
+        },
+        "requiredMembers": {
+          "description": "The private link resource required member names.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "requiredZoneNames": {
+          "description": "The private link resource Private link DNS zone name.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "PrivateLinkServiceConnectionState": {
+      "description": "A collection of information about the state of the connection between service consumer and provider.",
+      "type": "object",
+      "properties": {
+        "actionsRequired": {
+          "description": "Some RP chose \"None\". Other RPs use this for region expansion.",
+          "type": "string"
+        },
+        "description": {
+          "description": "User-defined message that, per NRP doc, may be used for approval-related message.",
+          "type": "string"
+        },
+        "status": {
+          "$ref": "#/definitions/PrivateEndpointServiceConnectionStatus"
+        }
+      }
+    },
+    "ProvisioningState": {
+      "description": "The current deployment state of workspace resource. The provisioningState is to indicate states for resource provisioning.",
+      "enum": [
+        "Unknown",
+        "Updating",
+        "Creating",
+        "Deleting",
+        "Succeeded",
+        "Failed",
+        "Canceled"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "PublicNetworkAccessType": {
+      "description": "The public network access flag.",
+      "enum": [
+        "Enabled",
+        "Disabled"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "PublicNetworkAccessType",
+        "modelAsString": true
+      }
+    },
+    "RaiBlocklistConfig": {
+      "description": "Azure OpenAI blocklist config.",
+      "type": "object",
+      "properties": {
+        "blocking": {
+          "type": "boolean"
```

Comment:
Add description? And ARM recommends string enum over boolean for many reasons. Consider something like activation: { Enabled, Disabled } (or similar with better names). Ref: https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review.

Reply:
I will add description for this, but we hope we can keep the boolean format as this schema is copied from this definition below and we want to keep the schema same so we can migrate the API.

https://github.com/Azure/azure-rest-api-specs/blob/ead20be2d8d680dfee3605e6f18148e65cbb79e4/specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2023-10-01-preview/cognitiveservices.json#L5604C10-L5604C18